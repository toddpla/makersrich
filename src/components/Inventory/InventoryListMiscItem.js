import React, { Component } from 'react';

class InventoryListMiscItem extends Component {

  changeMessage = () => {
      this.props.handleMessage(this.props.item.properties.message)
  }

  render() {
    return (
      <div className="inventory-list-item" onClick={this.changeMessage}>
        <div className={`miscellaneous-item ${this.props.item.properties.type}`}></div>
      </div>
    );
  }

}

export default InventoryListMiscItem;
