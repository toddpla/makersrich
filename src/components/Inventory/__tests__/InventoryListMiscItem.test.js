import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import InventoryListMiscItem from '../InventoryListMiscItem'
import { shallow } from 'enzyme'

const item = {
  type: 'miscellaneous',
  properties: {
    type: 'cheese',
    message: 'This is tasty'
  }
}

let handleMessage = jest.fn()

test('it renders a div with class matching the type of the item', () => {
  const renderer = new ShallowRenderer()

  renderer.render(<InventoryListMiscItem item={item}/>)
  const result = renderer.getRenderOutput()
  expect(result.props.children.props.className).toEqual('miscellaneous-item cheese');
})

test('#changeMessage changes message in inventory with item messsage', () => {
  let wrapper = shallow(
    <InventoryListMiscItem item={item} handleMessage={handleMessage}/>
  )
  wrapper.find('.inventory-list-item').simulate('click')
  expect(handleMessage).toHaveBeenLastCalledWith("This is tasty")
})

test('matches last snapshot', () => {
  let wrapper = shallow(
    <InventoryListMiscItem item={item} handleMessage={handleMessage}/>
  )
  expect(wrapper).toMatchSnapshot()
})
