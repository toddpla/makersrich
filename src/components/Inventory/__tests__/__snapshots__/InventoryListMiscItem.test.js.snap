// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches last snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InventoryListMiscItem
    handleMessage={
      [MockFunction] {
        "calls": Array [
          Array [
            "This is tasty",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    item={
      Object {
        "properties": Object {
          "message": "This is tasty",
          "type": "cheese",
        },
        "type": "miscellaneous",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="miscellaneous-item cheese"
      />,
      "className": "inventory-list-item",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "miscellaneous-item cheese",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="miscellaneous-item cheese"
        />,
        "className": "inventory-list-item",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "miscellaneous-item cheese",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
